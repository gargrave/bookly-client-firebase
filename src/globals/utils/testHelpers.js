import React from 'react';
import { shallow } from 'enzyme';

export class ComponentBuilder {
  constructor(component, defaultProps) {
    this.component = component;
    this.defaultProps = defaultProps;
  }

  shallowGetComponent(extraProps = {}) {
    const props = {
      ...this.defaultProps,
      ...extraProps,
    };
    return shallow(<this.component {...props} />);
  }
}
