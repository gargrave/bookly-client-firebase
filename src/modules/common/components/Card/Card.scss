@import '../../../../styles/_shadows';

$textColor: #6e6969;
$titleColor: #5d5656;

.card {
  @include boxShadowMed;
  background-color: #f9f9f9;
  border: solid 1px #bbb;
  border-radius: 5px;
  margin: 8px auto;
  max-width: 550px;
  padding: 10px 20px;
}

.hoverableCard:hover {
  background: #e4effc;
  cursor: pointer;
}

.header,
.title,
.text {
  margin: 0;
}

.header,
.title {
  color: $titleColor;
  font-weight: bold;
}

.header {
  font-size: 1.4rem;
}

.detailCard {
  padding: 15px;

  .title {
    font-size: 1.3rem;
    line-height: 2.5rem;
  }

  .title + .cardText {
    font-size: 1.1rem;
  }

  .cardText + .cardText {
    margin-top: 8px;
  }
}

.listDetailCard {
  padding: 10px
}
