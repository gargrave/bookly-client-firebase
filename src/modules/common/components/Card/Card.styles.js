import styled from 'react-emotion'
import { colors, shadows } from '../../../../styles/'

export const StyledWrapper = styled('div')`
  ${shadows.medium};

  background-color: ${colors.cardBg};
  border: solid 1px ${colors.cardBorder};
  border-radius: 5px;
  margin: 8px auto;
  max-width: 550px;
  padding: 10px 20px;

  &.hoverable:hover {
    background: ${colors.cardHoveredBg};
    cursor: pointer;
  }
`
