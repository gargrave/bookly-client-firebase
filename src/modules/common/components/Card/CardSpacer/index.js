import CardSpacer from './CardSpacer'

export default CardSpacer
