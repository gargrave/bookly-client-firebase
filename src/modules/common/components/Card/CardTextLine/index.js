import CardTextLine from './CardTextLine'

export default CardTextLine
