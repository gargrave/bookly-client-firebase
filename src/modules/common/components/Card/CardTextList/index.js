import CardTextList from './CardTextList'

export default CardTextList
