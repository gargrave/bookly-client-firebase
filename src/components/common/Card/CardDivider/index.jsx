// @flow
import React from 'react';

import { buildClasses } from '../../../../utils/cssHelpers';

type Props = {
};

const CardDivider = ({}: Props) => {
  return (
    <hr className={buildClasses(['card__divider'])} />
  );
};

CardDivider.propTypes = {
};

export default CardDivider;
