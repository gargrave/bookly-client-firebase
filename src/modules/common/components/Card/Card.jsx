// @flow
import React from 'react'
import { any, bool, func } from 'prop-types'
import styled from 'react-emotion'

import { buildClass } from '../../../../utils/cssHelpers'

import CardHeader from './CardHeader'
import CardSpacer from './CardSpacer'
import CardTextLine from './CardTextLine'
import CardTextList from './CardTextList'

import { colors, shadows } from '../../../../styles/'

type Props = {
  children?: any,
  hoverable?: boolean,
  onClick?: Function,
}

const Styled = styled('div')`
  ${shadows.medium};

  background-color: ${colors.cardBg};
  border: solid 1px ${colors.cardBorder};
  border-radius: 5px;
  margin: 8px auto;
  max-width: 550px;
  padding: 10px 20px;
  text-align: center;

  &.hoverable:hover {
    background: ${colors.cardHoveredBg};
    cursor: pointer;
  }
`

const Card = ({ children, hoverable = false, onClick }: Props) => {
  return (
    <Styled className={buildClass({ hoverable })} onClick={onClick}>
      {children}
    </Styled>
  )
}

Card.Header = CardHeader
Card.Spacer = CardSpacer
Card.TextLine = CardTextLine
Card.TextList = CardTextList

Card.propTypes = {
  children: any,
  hoverable: bool,
  onClick: func,
}

export default Card
