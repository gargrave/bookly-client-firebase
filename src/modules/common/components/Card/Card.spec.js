import { ComponentBuilder } from '../../../../utils/testHelpers'

import Card from './Card'

const defaultProps = {
  children: null,
  classes: [],
  header: 'Great Header',
  hoverable: true,
  onClick: jest.fn(),
  text: 'this is the text',
  title: 'Awesome Title',
}

const builder = new ComponentBuilder(Card, defaultProps)

describe('Card', () => {
  let component

  it('matches the snapshot', () => {
    component = builder.shallowGetComponent()
    expect(component).toMatchSnapshot()
  })

  // it('renders correctly', () => {
  //   component = builder.shallowGetComponent();
  //   expect(component.find('.bookly-card')).toHaveLength(1);
  //   expect(component.find('.bookly-card__header')).toHaveLength(1);
  //   expect(component.find('.bookly-card__title')).toHaveLength(1);
  //   expect(component.find('.bookly-card__text')).toHaveLength(1);
  //   expect(component.find('.bookly-card--hoverable')).toHaveLength(1);
  // });

  // it('does not render header when prop is blank', () => {
  //   component = builder.shallowGetComponent({ header: '' });
  //   expect(component.find('.bookly-card__header')).toHaveLength(0);
  // });

  // it('does not render title when prop is blank', () => {
  //   component = builder.shallowGetComponent({ title: '' });
  //   expect(component.find('.bookly-card__title')).toHaveLength(0);
  // });

  // it('does not render text when prop is blank', () => {
  //   component = builder.shallowGetComponent({ text: '' });
  //   expect(component.find('.bookly-card__text')).toHaveLength(0);
  // });

  // it('adds extra classes correctly', () => {
  //   component = builder.shallowGetComponent({ classes: ['oneclass', 'anotherclass'] });
  //   expect(component.find('.bookly-oneclass.bookly-anotherclass')).toHaveLength(1);
  // });

  // it('does not add a hover state with "hoverable=false"', () => {
  //   component = builder.shallowGetComponent({ hoverable: false });
  //   expect(component.find('.bookly--hoverable')).toHaveLength(0);
  // });

  it('calls the "onClick" callback when clicked', () => {
    component = builder.shallowGetComponent()
    expect(defaultProps.onClick.mock.calls.length).toBe(0)
    component.simulate('click')
    expect(defaultProps.onClick.mock.calls.length).toBe(1)
  })
})
