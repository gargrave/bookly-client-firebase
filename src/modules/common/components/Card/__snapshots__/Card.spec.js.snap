// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Card matches the snapshot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Card
    classes={Array []}
    header="Great Header"
    hoverable={true}
    onClick={[Function]}
    text="this is the text"
    title="Awesome Title"
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <p
          className={undefined}
        >
          Great Header
        </p>,
        <p
          className={undefined}
        >
          Awesome Title
        </p>,
        <p
          className={undefined}
        >
          this is the text
        </p>,
        null,
      ],
      "className": " ",
      "onClick": [Function],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Great Header",
          "className": undefined,
        },
        "ref": null,
        "rendered": "Great Header",
        "type": "p",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Awesome Title",
          "className": undefined,
        },
        "ref": null,
        "rendered": "Awesome Title",
        "type": "p",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "this is the text",
          "className": undefined,
        },
        "ref": null,
        "rendered": "this is the text",
        "type": "p",
      },
      null,
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <p
            className={undefined}
          >
            Great Header
          </p>,
          <p
            className={undefined}
          >
            Awesome Title
          </p>,
          <p
            className={undefined}
          >
            this is the text
          </p>,
          null,
        ],
        "className": " ",
        "onClick": [Function],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Great Header",
            "className": undefined,
          },
          "ref": null,
          "rendered": "Great Header",
          "type": "p",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Awesome Title",
            "className": undefined,
          },
          "ref": null,
          "rendered": "Awesome Title",
          "type": "p",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "this is the text",
            "className": undefined,
          },
          "ref": null,
          "rendered": "this is the text",
          "type": "p",
        },
        null,
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
